### PR TITLE
format_pre: Only run test for format check in CI

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: ci/format_pre.sh [hunk 1]
```diff
@@ -51,10 +51,11 @@ bazel run "${BAZEL_BUILD_OPTIONS[@]}" //configs:example_configs_validation
 CURRENT=spelling
 "${ENVOY_SRCDIR}"/tools/spelling/check_spelling_pedantic.py --mark check
 
-# TODO(phlax): move these to bazel rules
-CURRENT=check_format_test
-"${ENVOY_SRCDIR}"/tools/code_format/check_format_test_helper.sh --log=WARN
-
+# TODO(phlax): move clang/buildifier checks to bazel rules (/aspects)
+if [[ -n "$AZP_BRANCH" ]]; then
```
[the remark this answers is not in the document]
Comment (phlax) [reply ci/format_pre.sh:55]: this is a test for the test - unless the user is actually working on the test itself it shouldnt trigger and shouldnt really be run i think

these Prs are steps towards a goal of integrating the checks that end users need to run and making them performant enough to run in git hooks

this is a small opti as this codepath is currently being run by users and putting more tests together has slowed things down

[the remark this answers is not in the document]
Comment (adisuissa) [reply ci/format_pre.sh:55]: > this is a test for the test - unless the user is actually working on the test itself it shouldnt trigger and shouldnt really be run i think

Or if the user is working on `check_format` itself...

I'm good with approving this, as you are orchestrating most (if not all) of the work on this part, and this is not part of the code that changes often. Just stating that a safer way to make the change here is to add a flag that is called in CI, and the user can use the same flag.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

File: ci/format_pre.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (phlax) [reply ci/format_pre.sh:55]: the reason for not adding a flag (as in cli flag) is that it is run inside the docker container so means adding a load of code to get it there, when i think the safer/easier thing is just not to expose this to end users

until last week, this test was only being run in ci when a check failed, so this is kinda keeping the ~status quo

**Pull request merged**